lldb/source/Plugins/ObjectFile/Mach-O/ObjectFileMachO.cpp

```diff
@@ -1697,6 +1701,14 @@ void ObjectFileMachO::ProcessSegmentCommand(
     // isn't stored in the abstracted Sections.
     m_mach_sections.push_back(sect64);
 
+    // Make sure we can load dSYM files whose __DWARF sections exceed the 4GB
+    // barrier. llvm::MachO::section_64 have only 32 bit file offsets for the
+    // section contents.
+    const uint64_t section_file_offset = sect64.offset + section_offset_adjust;
+    // If this section overflows a 4GB barrier, then we need to adjust any
+    // subsequent the section offsets.
+    if (is_dsym && ((uint64_t)sect64.offset + sect64.size) >= UINT32_MAX)
+      section_offset_adjust += 0x100000000ull;
```

Comment:
```
    const uint64_t section_file_offset = sect64.offset + section_offset_adjust;
     if (is_dsym && ((uint64_t)sect64.offset + sect64.size) >= UINT32_MAX)
      section_offset_adjust += 0x100000000ull;
```

I wouldn't check `is_dsym`.  If this is correct, it is correct for any mach-o file that exceeds 4GB; the fact that we only see this in practice in dSYMs doesn't mean we should restrict it here.